### DISCUSSION TITLE
Enable the Github Wiki funktion

### DISCUSSION THREAD

**Discussion by Zoey2936** [nextcloud/all-in-one#826] [Ideas] [answered]
Enabling the GitHub wiki function would make the root of the repositories cleaner (then there would not be one instead of 6 .md files in the root of the repo) and the documentation files clearer and the documentation more structured

**Comment by szaimen** [comment, marked as answer]
Sorry but I disklike the wiki functiom of Github.
However it may be possible to put some docs into a docs folder...

**Reply by Zoey2936** [reply]
this is also possible, but I think the current structure of the docs is a bit unclear

**Reply by szaimen** [reply]
The  asic idea is currently that you search for something using the browsee search and the be redirected to the document itself, e.g. regarding reverse proxies...

**Comment by privatemaker** [comment]
@szaimen hello. i'm a bit confused- in #1945 I proposed 3 solutions to improve the documentation and offered to implement any. you closed my issue pointing to this issue. I see your comment:

>  However it may be possible to put some docs into a docs folder...

Is that to say my proposed number 2. idea is a welcome contribution? Pasting it here for clarity:

> Split up readme.md and migrate along with other .md files to static site like [docs.nextcloud.com](https://docs.nextcloud.com/)

The way I would do that would be in a `docs/` directory which could be then built into a static site. Please clarify your desired outcome(s) :smile: 

Given this issue is > 6 months old @Zoey2936 did you have interest in working on it or should I make an effort assuming it is welcome from @szaimen?

**Comment by szaimen** [comment]
Just to clarify and to give you some answers:

The problem that I see with the Guthub Wiki feature is that it is not possible to get PRs by the community.

Putting everything into a docs folder might make sense but then we will need to maintain the software that we are using for building a doc-site like e.g. sphinx.

Migrating things to docs.nextcloud.com is against this projects philosohpy of having everything related in one repo.

Using more details and summary tags makes things harder to find and not necessarily better readable.

**Comment by privatemaker** [comment]
> The problem that I see with the Guthub Wiki feature is that it is not possible to get PRs by the community.

Correct. A wiki can be cloned `git@github.com:nextcloud/all-in-one.wiki.git` but it is a separate repo with pull requests. I think you can specify who has access to edit it. Perhaps it *could* be committed to `docs/` folder in a dedicated branch and then a GH Action could auto-push on merges to a Wiki repo, but I haven't used GH actions so much.

> but then we will need to maintain the software that we are using for building a doc-site like e.g. sphinx.

Using Jekyll or Hugo (my preference) is not something requiring so much maintenance once setup, as both projects are rather mature and  straightforward.

> Migrating things to docs.nextcloud.com ...

I communicated wrong. I was not suggesting merge AIO Docs into the main NC Docs [repo](/nextcloud/documentation) but rather use the theme and style of the NC Docs so things are unified and easier to browse. Looking into that repo Sphinx is a bit less simple as what i'm used to (Hugo and the [learn-theme](https://learn.netlify.app/en/)) which can be built via Github Pages, but IIUC Sphinx can also be auto-built with this [sphinx-docs-to-github-pages](https://github.com/marketplace/actions/sphinx-docs-to-github-pages) action.

I'm not sure how it works with NC official things under the official `nextcloud/` github organization, but they might be up for making a sub-domain `example: all-in-one.nextcloud.com` or else it would just be `all-in-one.github.io` maybe @jancborchardt can clarify this part?

Anyway. A bit of discussion I guess :upside_down_face:

**Comment by Joly0** [comment]
https://www.growingwiththeweb.com/2016/07/enabling-pull-requests-on-github-wikis.html
Would this be a solution to the problem? That would enable users to be able to create pull requests to the wiki repo and those changes would be picked up by the wiki of the original repo.
The only "problem" i see here is, that it would require? a second repository for the wiki. But that would be the best solution in my opinion.

Its atleast by far easier to search the wiki, find stuff, etc when using the wiki function in github compared to the way it is done now.
